src/main/java/org/opensrp/repository/postgres/mapper/custom/xml/CustomLocationMetadataMapper.xml

```diff
@@ -217,5 +226,57 @@
 		LIMIT #{limit} OFFSET #{offset}
 
 	</select>
+	<sql id="searchCondition">
+        with locationAndTag as(
+        select
+        lm.id,
+        lm.location_id,
+        lm.parent_id,
+        lm."name",
+        ltm.location_tag_id,
+        lt."name" tag_name
+        from core.location_metadata as lm
+        join core.location_tag_map as ltm on lm.location_id = ltm.location_id
+        left join core.location_tag lt on lt.id = ltm.location_tag_id
+        <where>
+            <trim prefixOverrides="and">
+            <if test="locationSearchBean.name != null">
+				lm.name ilike '%${locationSearchBean.name}%'
+            </if>
+            <if test="locationSearchBean.locationTagId != null">
+                and ltm.location_tag_id = #{locationSearchBean.locationTagId}
+            </if>
+            <if test="locationSearchBean.parentId != null">
+                and lm.parent_id::bigint = #{locationSearchBean.parentId}
+            </if>
+            <if test="locationSearchBean.status != null">
+                and lm.status = #{locationSearchBean.status}
+            </if>
+            </trim>
+        </where>
+        ),
+        parent_location as (
+            select lt.parent_id,lt.location_id,lt.id,lt."name", lm."name" parent,
+            lt.tag_name from locationAndTag lt left join core.location_metadata lm
+            on lt.parent_id::bigint = lm.location_id
+        )
+    </sql>
+    <select id="selectLocations" resultMap="BaseCustomLocationResultMap">
+        <include refid="searchCondition"/>
+        select l.id,l.json - 'geometry' as json, pl.parent, pl.name,
```

Comment:
Since name and parent Id are available in location.properties, add the below to dynamically add location tag to location properties e.g `jsonb_set(json-'geometry','{properties,tag_name}','"asrt"',true)`

select l.id,jsonb_set(l.json-'geometry','{properties,tag_name}',to_jsonb(lt.name),true)`, pl.parent, pl.name